File: OOP/Projekt05__konsolowe-OOP__klasa-Player-zapis-do-pliku/python/main.py
import sys


class Player:
    def __init__(self, id, name, age, club):
        self.id = id
        self.name = name
        self.age = age
        self.club = club

def main():
    try:
        players: list[Player] = [
            Player(1, "Paulo", 20, "FC Potato"),
            Player(2, "Karl", 30, "FC Tomato"),
            Player(3, "Grzegorz", 25, "FC Polako"),
            Player(4, "John", 24, "FC Uralto"),
            Player(5, "Fabiano", 22, "FC Utarto"),
        ]

        fileName: str

        if len(sys.argv) < 2:
            fileName = input(
                "Podaj nazwe pliku do ktorego chcesz zapisac dane (z .txt): "
            )
        else:
            fileName = sys.argv[1]

        with open(fileName, "w") as f:
            for player in players:
                f.write(f"Gracz o ID: {player.id}\n")
                f.write(f"{player.name}\n")
                f.write(f"{player.age}\n")                
                f.write(f"{player.club}\n")                
                f.write("\n")
    except Exception as ex:
        print(ex)


if __name__ == "__main__":
    main()

File: OOP/Projekt05__konsolowe-OOP__klasa-Player-zapis-do-pliku/cpp/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cstdio>

using namespace std;

class Player
{
public:
    unsigned int id;
    string name;
    unsigned int age;
    string club;

    Player(unsigned int id, string name, unsigned int age, string club)
    {
        this->id = id;
        this->name = name;
        this->age = age;
        this->club = club;
    }
};

int main(const int argc, const char *argv[])
{
    Player players[5] = {Player(1, "Paulo", 20, "FC Potato"), Player(2, "Karl", 30, "FC Tomato"), Player(3, "Grzegorz", 25, "FC Polako"), Player(4, "John", 24, "FC Uralto"), Player(5, "Fabiano", 22, "FC Utarto")};
    string fileName;

    if (argc < 2)
    {
        cout << "Nazwe mozna przekazac jako parametr programu: app.exe <fileName>" << endl;
        cout << "Podaj nazwe pliku do ktorego chcesz zapisac dane (z .txt): ";
        cin >> fileName;
    }
    else
        fileName = argv[1];

    try
    {
        ofstream File(fileName);
        for (int i = 0; i < 5; i++)
        {
            Player player = players[i];

            File << "Gracz o ID: " << player.id << endl;
            File << player.name << endl;
            File << player.age << endl;
            File << player.club << endl;
            File << endl;
        }
    }
    catch (exception &ex)
    {
        cout << ex.what() << endl;
    }

    return 0;
}